### PR TITLE
refactor: refactored strategies to use the template design pattern

### PR DESCRIPTION
**Pull request by liorfr**

### CODE REVIEW

File: test/appstudio_strategy.spec.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import { expect } from "chai";
```
Comment (allacherkes) [inline test/appstudio_strategy.spec.ts:1]: rename the file to strategy.spec.ts as it is testing several files inside strategy folder

**Review by michawai — APPROVED** [review]

File: src/strategy/appStudioStrategies.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  // Check if the user has any restrictions defined in the strategy (defined in the FT server)
+  abstract isDefined(parameters: AppStudioMultiParameters): boolean;
+
+  // context.currentEnvironment - The Environment that the extension is running on (calculated in this client)
+  // parameters.environments // The Environments that the extension is allowed to be enabled on (list is defined on the server)
+  abstract isEnabled(parameters: AppStudioMultiParameters, context: AppStudioMultiContext): boolean;
+}
+
+export function registerStrategies(appStudioMultiStrategy: AppStudioMultiStrategy): void {
```
[the remark this answers is not in the document]
Comment (liorfr) [reply src/strategy/appStudioStrategies.ts:16]: `AppStudioMultiStrategy` is a generic class that should not know the concrete implementation.
`unleash_client_manager.ts` manage `AppStudioMultiStrategy`, initialize and register the strategies

File: src/unleash_client_manager.ts [hunk 3]
```diff
@@ -13,7 +14,9 @@ async function createNewUnleashClient(extensionName: string, unleashClientMap: M
   const serverArgs = ServerArgs.getServerArgs();
 
   //init client
-  const client = await initializeUnleashClient(extensionName, serverArgs, [new AppStudioMultiStrategy()]);
+  const appStudioMultiStrategy = new AppStudioMultiStrategy();
+  const client = await initializeUnleashClient(extensionName, serverArgs, [appStudioMultiStrategy]);
+  registerStrategies(appStudioMultiStrategy);
```
[the remark this answers is not in the document]
Comment (liorfr) [reply src/unleash_client_manager.ts:19]: `AppStudioMultiStrategy` is a generic class that should not know the concrete implementation.
`unleash_client_manager.ts` manage `AppStudioMultiStrategy`, initialize and register the strategies

File: test/appstudio_strategy.spec.ts (hunk 1 above)
Comment (liorfr) [reply test/appstudio_strategy.spec.ts:1]: done

**Pull request merged**